staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -243,6 +251,33 @@ func (c completedConfig) NewWithDelegate(delegationTarget genericapiserver.Deleg
 		go availableController.Run(5, context.StopCh)
 		return nil
 	})
+	s.GenericAPIServer.AddPostStartHook("apiservice-wait-for-first-sync", func(context genericapiserver.PostStartHookContext) error {
```

Comment:
@deads2k compare https://github.com/kubernetes/kubernetes/pull/89147. That looks much cleaner and simpler.

Reply:
Done. Added UPSTREAM: 89147: aggregator: wait for complete proxy handler prior to UPSTREAM: <carry>: wait for apiservices on startup